src/components/data/EntitiesContent.js

```diff
@@ -359,6 +359,87 @@ const EntitiesContent = ({
     ])])
   }
 
+  const entitiesSelected = !_.isEmpty(selectedEntities)
+  const canEdit = !Utils.editWorkspaceError(workspace)
+
+  const renderEditMenu = () => {
+    return !snapshotName && h(MenuTrigger, {
+      side: 'bottom',
+      closeOnClick: true,
+      content: h(Fragment, [
+        h(MenuButton, {
+          onClick: () => setDeletingEntities(true)
+        }, 'Delete')
+      ])
+    }, [h(ButtonSecondary, {
+      disabled: !canEdit || !entitiesSelected,
+      tooltip: Utils.cond(
+        [!canEdit, () => 'You do not have permission to modify this workspace'],
+        [!entitiesSelected, () => 'Select rows to edit'],
+        () => 'Edit selected data'
+      ),
+      style: { marginRight: '1.5rem' }
+    }, [icon('edit', { style: { marginRight: '0.5rem' } }), 'Edit'])])
+  }
+
+  const renderExportMenu = ({ columnSettings }) => {
+    const isSet = _.endsWith('_set', entityKey)
+    const isSetSet = entityKey.endsWith('_set_set')
+
+    return h(MenuTrigger, {
+      side: 'bottom',
+      closeOnClick: true,
+      content: h(Fragment, [
+        h(MenuButton, {
+          disabled: isSetSet,
+          tooltip: isSetSet && 'Downloading sets of sets as TSV is not supported at this time',
+          onClick: async () => {
+            const tsv = buildTSV(columnSettings, selectedEntities)
+            isSet ?
+              FileSaver.saveAs(await tsv, `${entityKey}.zip`) :
+              FileSaver.saveAs(new Blob([tsv], { type: 'text/tab-separated-values' }), `${entityKey}.tsv`)
+            Ajax().Metrics.captureEvent(Events.workspaceDataDownloadPartial, {
+              ...extractWorkspaceDetails(workspace.workspace),
+              downloadFrom: 'table data',
+              fileType: '.tsv'
+            })
+          }
```

Comment:
Since this block is identical to the one defined [above](https://github.com/DataBiosphere/terra-ui/pull/2907/files#diff-169a9d1c74e376c576f00ab861b0f13ecbb91370660668f3f61d9bf5193e5bcbR331), I wonder if it'd be worth factoring it out.

Reply:
Done in [2f1064e](https://github.com/DataBiosphere/terra-ui/pull/2907/commits/2f1064ec5e5b161b3a7f94074b837f084338e02a).